From 4fa7de1eb28d97837c64578c6e5f6ff835e548e4 Mon Sep 17 00:00:00 2001
From: Dongil Park <dongil.park@lge.com>
Date: Fri, 15 Dec 2017 10:50:44 +0900
Subject: playbin3: Fix accessing invalid index in GstStream when received
 select-stream event

If select-stream event was send to playbin3 as missing any GstStream of ES type
(V or A or TEX) of collection then, playbin will access to invalid address of
GstStream due to invalid index limit. This caused SIGSEGV.

https://bugzilla.gnome.org/show_bug.cgi?id=791638
---
 gst/playback/gstplaybin3.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gst/playback/gstplaybin3.c b/gst/playback/gstplaybin3.c
index f3f5efe..36135cb 100644
--- a/gst/playback/gstplaybin3.c
+++ b/gst/playback/gstplaybin3.c
@@ -2252,7 +2252,7 @@ get_combiner_stream_id (GstPlayBin3 * playbin, GstSourceCombine * combine,
   gint i;
   GList *tmp;
 
-  for (i = 0; combine->streams->len; i++) {
+  for (i = 0; i < combine->streams->len; i++) {
     GstStream *stream = (GstStream *) g_ptr_array_index (combine->streams, i);
     const gchar *sid = gst_stream_get_stream_id (stream);
     for (tmp = full_list; tmp; tmp = tmp->next) {
-- 
cgit v1.1

